ta4j-core/src/main/java/org/ta4j/core/BaseTradingRecord.java

```diff
@@ -53,6 +53,12 @@ this software and associated documentation files (the "Software"), to deal in
 
     /** The entry type (BUY or SELL) in the trading session */
     private Order.OrderType startingType;
+
+    /** The starting index of the trading session */
+    private int startIndex;
+
+	/** The finishing index of the trading session */
+    private int finishIndex;
```

Comment:
- These fields are never used locally and just needed to clarify what start and end of this trading record is, or? There are two constructors that don't initialize the fields. -> initialize them or handle it in the getters (example in getter comment)

- The `TimeSeriesManager` creates `new BaseTradingRecord` with corresponding **startIndex** and **finishIndex**  and **just** records orders that are **between** **startIndex** and **finishIndex** we deliver in `TimeSeriesManager#run(order, amount, startIndex, endIndex) ` method -> that looks valid.
But are there other situation in which an order can be insert into `BaseTradingRecord` that is out of scope (not between **startIndex** and **finishIndex**)? I could imagine in a custom extend of `TimeSeriesManager` one could try to do so. Should this be possible? I think this would lead to an inconsistence state (a `BaseTradingRecord` holds orders that are out of its range(startIndex, finishIndex)) that we should not allow. -> Maybe we could use those fields to check in `enter(index, price, amount)` and `exit(index, price, amount)` if the order is in scope?

Reply:
Other possibility would be to remove `startIndex` and `finishIndex` fields and do not let the `BaseTradingRecord` decide which orders are valid or not. But I would prefer your solution with `startIndex` and `finishIndex` fields, because it is more transparent which frame this records covers.

Reply:
> These fields are never used locally and just needed to clarify what start and end of this trading record is

Yes, and the new indices are only meaningful if the new constructor is used.  I will add inits of 0 and Integer.MAX_SIZE to the old constructor(s).  The new constructor indices are simply stored and do not internally impact logic in any way.  My only current use of the indices is for calls like this first one
```
buyAndHold.calculate(series, tradingRecord, tradingRecord.getStartIndex(), tradingRecord.getFinishIndex()); // new, limit range of series
buyAndHold.calculate(series, tradingRecord, currentIndex-199, currentIndex); // new, limit range to previous 200 indices
buyAndHold.calculate(series, tradingRecord); // old behavior, use entire range of series
```
Also, one could use the old constructors but evaluate VersusBuyAndHold for the previous 200-tick windows like the second above.  CAVEAT: that probably requires better handling of trades spanning across either index (like fake openOrder at startIndex and closeOrder at finishIndex.)  But that is all new functionality that won't break anything existing as it gets worked out.

So, the constructor indices are not coupled to the calculate() indices unless you couple them with the getters as above.

> an order can be insert into BaseTradingRecord that is out of scope

I think that's fine.  To me, scope is defined by the call to calculate(), not by the record constructor.  Add orders outside of startIndex and finishIndex.  Call the third example above and they are included.  Call the first example and they are excluded (but see the CAVEAT.)  In fact, I'll probably add these cases to the unit test to demonstrate.

Finally, if your series manager calls the new constructor with indices inside run(), then it's your responsibility to only operate() between those indices.  You don't have to keep them locally in run() though, as you can then use the getters.  If you operate() outside of your subsequent calculate() indices (or calculate() outside your operate() indices), you are probably just wasting cycles.